tests/test_jobs.py

```diff
@@ -2553,7 +2553,7 @@ def test_dagrun_root_after_dagrun_unfinished(self):
         ti_ids = [(ti.task_id, ti.state) for ti in first_run.get_task_instances()]
 
         self.assertEqual(ti_ids, [('current', State.SUCCESS)])
-        self.assertEqual(first_run.state, State.SUCCESS)
+        self.assertIn(first_run.state, [State.SUCCESS, State.RUNNING])
```

Comment:
@feng-tao may be necessary to update the docstring of this test case as well? Currently it's `DagRuns with one successful and one future root task -> SUCCESS`

Reply:
I think in real scenario it will only have success state for the task. The issue only happens in unit test as we don't wait / sleep to retrieve the result during CI test for speeding up the CI test effort hence we sometimes(or in some ORM) get the running state for the task

Reply:
updated with a note.

Reply:
Not sure if I'm missing anything, but I'm thinking if it would be better/clearer to just add a sleep here If the race condition only occurs in this test case?

Reply:
@XD-DENG , you could take a look at more on this pr(https://github.com/apache/airflow/pull/4730/files) as it tries to speed up the CI time.

Reply:
Very similar to https://github.com/apache/airflow/pull/4737/files

Reply:
Thanks @feng-tao for the clarification.